app/stores/ada/TrezorConnectStore.js

```diff
@@ -321,58 +324,62 @@ export default class TrezorConnectStore extends Store {
     this.progressInfo.currentStep = ProgressStep.SAVE;
     this.progressInfo.stepState = StepState.PROCESS;
 
-    if (this.trezorDeviceInfo
-      && this.trezorDeviceInfo.publicKey
-      && this.trezorDeviceInfo.features) {
-      const walletData = {
-        walletName,
-        publicMasterKey: this.trezorDeviceInfo.publicKey,
-        deviceFeatures: this.trezorDeviceInfo.features
-      };
-      this._saveTrezor(walletData);
-    }
-  };
-
-  @action _goToSaveError = (): void => {
-    this.progressInfo.currentStep = ProgressStep.SAVE;
-    this.progressInfo.stepState = StepState.ERROR;
+    this._saveHW(walletName);
   };
 
   /** creates new wallet and loads it */
-  _saveTrezor = async (params: {
-    publicMasterKey: string,
-    walletName: string,
-    deviceFeatures: Features,
-  }): Promise<void>  => {
+  _saveHW = async (walletName: string): Promise<void>  => {
     try {
-      Logger.debug('TrezorConnectStore::_saveTrezor:: stated');
-      this._setIsCreateTrezorWalletActive(true);
-      this.createTrezorWalletRequest.reset();
+      Logger.debug('TrezorConnectStore::_saveHW:: stated');
+      this._setIsCreateHWActive(true);
+      this.createHWRequest.reset();
+
+      const reqParams = this._prepareCreateHWReqParams(walletName);
+      const trezorWallet: Wallet =
+        await this.createHWRequest.execute(reqParams).promise;
 
-      const trezorWallet: Wallet = await this.createTrezorWalletRequest.execute(params).promise;
       if (trezorWallet) {
         await this._onSaveSucess(trezorWallet);
       } else {
-        // this Error will be converted to LocalizableError()
-        throw new Error();
+        throw new UnexpectedError();
       }
     } catch (error) {
+      Logger.error(`TrezorConnectStore::_saveHW::error ${stringifyError(error)}`);
+
       if (error instanceof CheckAdressesInUseApiError) {
-        // redirecting CheckAdressesInUseApiError -> saveError101
-        // because for user saveError101 is more meaningful in this context
-        this.error = new LocalizableError(messages.saveError101);
+        // redirecting CheckAdressesInUseApiError -> hwConnectDialogSaveError101
+        // because for user hwConnectDialogSaveError101 is more meaningful in this context
+        this.error = new LocalizableError(globalMessages.hwConnectDialogSaveError101);
       } else if (error instanceof LocalizableError) {
         this.error = error;
       } else {
         // some unknow error
-        this.error = new LocalizableError(messages.error999);
+        this.error = new UnexpectedError();
```

Comment:
that is not swallow, it has fixed message inside it.
UnexpectedError is a LocalizableError for UI display (in case we are not sure which error happened)